packages/flutter/lib/src/widgets/form.dart

```diff
@@ -200,8 +227,8 @@ class _FormScope extends InheritedWidget {
     required super.child,
     required FormState formState,
     required int generation,
-  }) : _formState = formState,
-       _generation = generation;
+  })  : _formState = formState,
```

Comment:
Can you undo these formatting changes?  I think it should be the way it was if I'm not mistaken.  Here and elsewhere.

Reply:
> @casvanluijtelaar I've modified your gist to show what you'd have to do to achieve (mostly) the same thing without this PR: https://gist.github.com/justinmc/1ed6f6a2faa3bf5dc0a6ce83119afeeb
> 
> Does that cover your use case or am I missing anything? From that it looks like the hardest part is just getting access to all of the FormFields that belong to the Form. In my example I had to keep track of all of their keys myself.
> 
> What if we just made that part easier by exposing all of the FormFieldStates or their keys on FormState? I kind of wonder why that wasn't done originally when Form was written...

more or less, yes. this PR tries to take advantage of the formSates ability to interact with all child Form Fields. without having to to the boilerplate of keeping track of them yourself. My original idea was to just do this as a method extension, but since the _fields aren't publically available I had to clone the form implementation and add it myself. hence the PR ;)

Reply:
and as said, your example would scale a lot harder with a more complex UI.